selfdrive/controls/controlsd.py

```diff
@@ -168,37 +169,37 @@ def state_transition(frame, CS, CP, state, events, soft_disable_timer, v_cruise_
 
   # ENABLED
   elif state == State.enabled:
-    if get_events(events, [ET.USER_DISABLE]):
+    if has_event(events, [ET.USER_DISABLE]):
       state = State.disabled
       AM.add(frame, "disable", enabled)
 
-    elif get_events(events, [ET.IMMEDIATE_DISABLE]):
+    elif has_event(events, [ET.IMMEDIATE_DISABLE]):
       state = State.disabled
       for e in get_events(events, [ET.IMMEDIATE_DISABLE]):
```

Comment:
The same lookup is done twice here. Can we combine them in a smart way? Maybe convert the events into a dict first of event type to list of events of that type.

Reply:
In general there is a lot of repeated code, is it possible to abstract some of it into a function.

Reply:
I have added a set to do this. dict will create lists for types everytime we might not need.